lua/metals.lua

```diff
@@ -129,15 +129,16 @@ end
 
 M.logs_toggle = function()
   local bufs = api.nvim_list_bufs()
-  for _, v in ipairs(bufs) do
+  for i, v in ipairs(bufs) do
     local buftype = api.nvim_buf_get_option(v, 'buftype')
-    if buftype == 'terminal' then
-      log.info_and_show('Logs are already opened. Try an :ls to see where it is.')
+    if buftype == 'terminal' and vim.b.metals_buf_purpose == 'logs' then
+      log.info_and_show(string.format('Logs are already opened. Try an :ls to see where it is.' ..
+                                          'Or switch to it directly using :b%i', i))
       return
     end
   end
   -- Only open them if a terminal isn't already open
-  api.nvim_command [[vsp term://tail -f .metals/metals.log]]
+  api.nvim_command [[vsp term://tail -f .metals/metals.log | :let b:metals_buf_purpose="logs"]]
```

Comment:
Also this is a bit of a nit, but I try to avoid doing to much in a command like this, so we can also separate this out.
```suggestion
  api.nvim_command [[vsp term://tail -f .metals/metals.log]]
  vim.b.metals_buf_purpose = 'logs'
```

Reply:
```
    lua/metals.lua:145:3: setting read-only field b.metals_buf_purpose of global vim
```

🙃 

Is there a way to make this check more fine-grained?

Reply:
You an just add it to `globals` in the `.luacheckrc` file.